src/Nethermind/Nethermind.Blockchain/BeaconBlockRoot/BeaconBlockRootHandler.cs

```diff
@@ -15,25 +15,42 @@ public class BeaconBlockRootHandler(ITransactionProcessor processor) : IBeaconBl
 {
     private const long GasLimit = 30_000_000L;
 
-    public void StoreBeaconRoot(Block block, IReleaseSpec spec)
+    public (Address? toAddress, AccessList? accessList) BeaconRootsAccessList(Block block, IReleaseSpec spec)
     {
         BlockHeader? header = block.Header;
-        var canInsertBeaconRoot = spec.IsBeaconBlockRootAvailable
+        bool canInsertBeaconRoot = spec.IsBeaconBlockRootAvailable
                                   && !header.IsGenesis
                                   && header.ParentBeaconBlockRoot is not null;
 
-        if (canInsertBeaconRoot)
+        Address? eip4788ContractAddress = canInsertBeaconRoot ?
+            spec.Eip4788ContractAddress ?? Eip4788Constants.BeaconRootsAddress :
+            null;
+
+        return (eip4788ContractAddress,
+            eip4788ContractAddress is null ?
+                null :
+                new AccessList.Builder()
+                    .AddAddress(eip4788ContractAddress)
+                    .AddStorage(block.Timestamp % 8191)
+                    .Build());
+    }
+
+    public void StoreBeaconRoot(Block block, IReleaseSpec spec)
+    {
+        (Address? toAddress, AccessList? accessList) = BeaconRootsAccessList(block, spec);
+
+        if (toAddress is not null)
         {
-            Address beaconRootsAddress = spec.Eip4788ContractAddress ?? Eip4788Constants.BeaconRootsAddress;
+            BlockHeader? header = block.Header;
             Transaction transaction = new()
             {
                 Value = UInt256.Zero,
                 Data = header.ParentBeaconBlockRoot.Bytes.ToArray(),
-                To = beaconRootsAddress,
+                To = toAddress,
                 SenderAddress = Address.SystemUser,
                 GasLimit = GasLimit,
                 GasPrice = UInt256.Zero,
-                AccessList = new AccessList.Builder().AddAddress(beaconRootsAddress).Build()
+                AccessList = accessList
```

Comment:
While it might be good idea to add `block.Timestamp % 8191` as storage to access list, Geth is not doing that. And if Geth is not doing that than an edge case can happen where geth transaction goes out of gas and ours don't (in general it shouldn't happen with current gas limit but still) - and that would result in consensus issue.

Reply:
Removed it for the tx access list